File: 27.js
var removeElement = function (nums, val) {
  let n = nums.length;
  let k = 0;
  let left = 0;
  let right = n - 1;
  while (left <= right) {
    if (nums[left] === val && nums[right] !== val) {
      [nums[left], nums[right]] = [nums[right], nums[left]];
      left++;
      right--;
    }
    if (nums[left] !== val) {
      k++;
      left++;
    }
    if (nums[right] === val) {
      right--;
    }
  }
  return left;
};

File: 27.py
def removeElement(nums, val):
        i=0
        j=len(nums)-1
        # res=[]
        while i<j:
            if nums[j]==val:
                while nums[j]==val:
                    j-=1
            if nums[i]==val:
                nums[i],nums[j]=nums[j],nums[i]
                i+=1
                j-=1
            else: 
                i+=1
        # for i in range(0,i):
        #     res.append(nums[i])
        # i =1
        return i
# Example usage
print(removeElement([3,2,2,3],3))   